### PR TITLE
Update istio test reference to latest commit and fix tests

### PR DESCRIPTION
**Pull request by ericvn**
## Description

Update istio test reference to latest commit and fix tests

Supersedes https://github.com/istio/istio.io/pull/14184

Test fixes:

https://github.com/istio/istio/pull/47860 causes the operator test to fail trying to delete the Istio-system namespace with a IstioOperator.  Added steps to delete the two istiooperators to prevent the namespace deletion hang. https://github.com/istio/istio/pull/48106 should fix this and I will remove the changes I made once that PR merges (I'll update to that commit).

The PR also caused an issue in the egress-control test as it used installed-state. The test was rewritten to not use installed-state.

https://github.com/istio/istio/pull/47922 updated the port for httpbin from 80->8080, so this was updated in authn-policy.

https://github.com/istio/istio/pull/47407 causes the jwt-route tests to fail. This will require a change in the upstream jwt filter (https://github.com/envoyproxy/envoy/pull/31091). Will move to that commit once it makes it way into istio.isito.

https://github.com/istio/common-files/pull/953 updates the metallb version. The new version creates a validatingwebhook in the cluster which affects some tests. Need to ignore the new webhook and also not delete it in one test (the test blindly deletes all web hooks).

tasks/observability/logs/telemetry-api/test.sh is failing. Test is disabled and https://github.com/istio/istio.io/issues/14210 was opened to make sure we enable it again.

### CODE REVIEW

**Review by dhawton — APPROVED** [review]

**Comment by ericvn** [comment]
/test lint

flake

**Comment by ericvn** [comment]
Default: TestDocs/tasks/security/authentication/jwt-route/test.sh - Noted above, likely caused by https://github.com/istio/istio/pull/47407.

**Comment by ericvn** [comment]
@jwendell Test failures due to the add/removal of the metallb webhook. Assuming this is due to the common-files change.

**Comment by ericvn** [comment]
mutlicluster:
- setup/install/external-controlplane/gtwapi_test.sh and test.sh
```
VERIFY FAILED snip_set_up_the_remote_config_cluster_5 (timeout after 120s):
received:
"NAME                              WEBHOOKS   AGE
istio-validator-external-istiod   1          2m2s
istiod-default-validator          1          2m2s
metallb-webhook-configuration     7          11m"
expected:
"NAME                              WEBHOOKS   AGE
istio-validator-external-istiod   1          6m53s
istiod-default-validator          1          6m53s"
```

**Comment by ericvn** [comment]
Default:
- tasks/security/authorization/authz-ingress/gtwapi_test.sh -- Flake?
- tasks/security/authentication/jwt-route/test.sh -- Known

**Comment by ericvn** [comment]
None:
- tasks/observability/logs/telemetry-api/test.sh (same as above)
```
VERIFY FAILED count_httpbin_pod (timeout after 120s):
received:
"39"
expected:
"10"
```
- tasks/traffic-management/egress/egress-control/test.sh (same as above)
```
VERIFY FAILED snip_envoy_passthrough_to_external_services_1 (timeout after 120s):
received:
"error: the server doesn't have a resource type "istiooperator""
expected:
"REGISTRY_ONLY"
```

**Comment by ericvn** [comment]
> * tasks/observability/logs/telemetry-api/test.sh (same as above)

https://github.com/istio/istio.io/issues/14210 opened to re-enable the test which is disabled not.

**Comment by ericvn** [comment]
I disabled the jwt-route test sine the Envoy PR is not merged and opened a release blocker issue https://github.com/istio/istio.io/issues/14224.

File: content/en/docs/ops/diagnostic-tools/istioctl-analyze/test.sh [hunk 1]
```diff
@@ -103,7 +103,7 @@ kubectl label namespace default istio-injection-
 kubectl delete ns frod
 kubectl delete deployment my-deployment
 kubectl delete vs ratings
-# Delete the Istio this test installed
-kubectl get validatingwebhookconfigurations -o custom-columns=NAME:.metadata.name --no-headers | xargs kubectl delete validatingwebhookconfigurations
+# Delete the Istio this test installed - Don't remove the metallb validatingwebhookconfiguration
+kubectl get validatingwebhookconfigurations -o custom-columns=NAME:.metadata.name --no-headers | grep -v metallb | xargs kubectl delete validatingwebhookconfigurations
 kubectl get mutatingwebhookconfigurations -o custom-columns=NAME:.metadata.name --no-headers | xargs kubectl delete mutatingwebhookconfigurations
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl-analyze/test.sh:108]: This uninstall seems kind of weird (and fragile!). Why not just uninstall?
```suggestion
# Delete the Istio this test installed
istioctl uninstall -y --purge
```

**Comment by frankbu** [comment]
Empty file?

image

**Comment by ericvn** [comment]
> Empty file?
> 
> image

It's picking on the chmod of the file and there are no file changes. I can update that as well.

**Pull request merged**